packages/connectors-web/src/alchemySmartWallet.ts

```diff
@@ -1,85 +1,333 @@
+ * @returns {CreateConnectorFn} The Alchemy Smart Wallet connector, which can be passed to Wagmi's `createConfig`.
+ */
+export function alchemySmartWallet(
+  options: AlchemySmartWalletOptions,
+): CreateConnectorFn<Provider, Properties> {
   return createConnector((config) => {
-    const baseConnector = createBaseConnector(config);
+    // `config` contains an event emitter that all of Wagmi listens to. Don't
+    // let the inner connector get its hands on it: we want to fully encapsulate
+    // the inner connector so it doesn't affect global state. Note that the uid
+    // doesn't matter since this emitter is internal to this connector instance.
+    const innerEmitter = new Emitter("inner-uid");
+    const innerConnector = options.ownerConnector({
+      ...config,
+      emitter: innerEmitter,
+    });
+
+    /*
+     * We'll return this useless provider when the smart wallet connector is not
+     * connected. We need to return _something_ in response to `getProvider`
+     * even when not connected, or Wagmi will skip over this connector.
+     *
+     * TODO: Should we return the same provider instance every time, with that
+     * instance being a proxy object routing to either this or the real provider
+     * if it exists? We would need to do that if Wagmi held on to a provider
+     * that it first got during initialization. It doesn't seem to do that
+     * today, but what if it changed behavior?
+     */
+    const loggedOutProvider: Provider = {
+      on: () => {},
+      removeListener: () => {},
+      request: async (requestParams) => {
+        if (requestParams.method === "eth_accounts") {
+          return [] as any;
+        }
+        const baseProvider = (await innerConnector.getProvider()) as any;
+        return baseProvider.request(requestParams);
+      },
+    };
+
+    let smartWalletClientPromise: Promise<SmartWalletClient> | undefined;
+    let currentChainId: number | undefined;
+    let currentInnerAccount: Address | undefined;
+
+    const resetState = (): void => {
+      smartWalletClientPromise = undefined;
+      currentChainId = undefined;
+      currentInnerAccount = undefined;
+    };
+
+    const createClient = async (): Promise<SmartWalletClient> => {
+      const chainId = await innerConnector.getChainId();
+      const innerAccounts = await innerConnector.getAccounts();
+      if (innerAccounts.length === 0) {
+        throw new Error("No accounts found in base connector");
+      }
+      // TODO: do we always take the first account?
+      const signerAddress = innerAccounts[0];
+      const chain = getChainFromConfig(config, chainId);
+      const provider = (await innerConnector.getProvider()) as EIP1193Provider;
+      const signer = createWalletClient({
+        account: signerAddress,
+        chain,
+        transport: custom(provider),
+      });
```

Comment:
❗ if `innerConnector.getClient` is defined, we should use that to get the signer client. using `innerClient.getProvider()` and wrapping it in a new client will not allow signing 7702 authorizations.